### DISCUSSION TITLE
Episode Scrobble ID order of preference?

### DISCUSSION THREAD

**Discussion by SenorSmartyPants** [trakt/trakt-api#324] [Q&A] [answered]
I've done some work on the Jellyfin trakt plugin. How are episodes matched when not all the ids provided in a scrobble match what trakt has on file? This is related to the most recent 2 Doctor Who Specials.

I have 'Revolution of the Daleks' numbered matching tvdb, 0x157, but Trakt uses TMDB number 0x158. I also supply TVDB and IMDB ids in my media object for the scrobble.

`{ show: { title: "Doctor Who (2005)", year: 2005, ids: { tvdb: 78804, tvrage: 3332, imdb: "tt0436992", tmdb: 57243, trakt: null, slug: null } }, episode: { season: 0, number: 157, title: null, ids: { tvdb: 7615160, tvrage: null, imdb: "tt11892730", tmdb: null, trakt: null, slug: null } }`

But this results in matching 0x157 "Farewell, Sarah Jane" on trakt. 

I had hoped providing all ids I had would result in a good match. JF-Trakt used to only provide TVDBid if JF had one, but this resulted in missed scrobbles when Trakt didn't have TVDBid. I changed the plugin in to provide all external ids and season/episode numbers. I had been working well until this episode.

**Comment by rectifyer** [comment, marked as answer]
https://trakt.docs.apiary.io/#reference/sync

The 2nd paragraph in `Media objects for syncing` has the order we use for finding media objects.

> Media objects will be matched by ID first, then fall back to title and year. IDs will be matched in this order `trakt`, `imdb`, `tmdb`, `tvdb`, and `slug`. If nothing is found, it will match on the `title` and `year`. If still nothing, it would use just the `title` (or `name` for people) and find the most current object that exists.

**Reply by SenorSmartyPants** [reply]
This does not address episode numbers (season + episode) versus IDs. And from my testing in January trakt seems to use season/episode before episode IDs

**Reply by rectifyer** [reply]
If you send seasons + episodes under the show object, it will lookup on the season number and episode number only.

```
"shows": [
    {
      "title": "Breaking Bad",
      "year": 2008,
      "ids": {
        "trakt": 1,
        "slug": "breaking-bad",
        "tvdb": 81189,
        "imdb": "tt0903747",
        "tmdb": 1396
      }
    },
    {
      "title": "The Walking Dead",
      "year": 2010,
      "ids": {
        "trakt": 2,
        "slug": "the-walking-dead",
        "tvdb": 153021,
        "imdb": "tt1520211",
        "tmdb": 1402
      },
      "seasons": [
        {
          "watched_at": "2014-09-01T09:10:11.000Z",
          "number": 3
        }
      ]
    },
    {
      "title": "Mad Men",
      "year": 2007,
      "ids": {
        "trakt": 4,
        "slug": "mad-men",
        "tvdb": 80337,
        "imdb": "tt0804503",
        "tmdb": 1104
      },
      "seasons": [
        {
          "number": 1,
          "episodes": [
            {
              "watched_at": "2014-09-01T09:10:11.000Z",
              "number": 1
            },
            {
              "number": 2
            }
          ]
        }
      ]
    }
  ]
```

If you want to use ID lookups only, it should be sent in this format:

```
"seasons": [
    {
      "watched_at": "2019-01-02T09:10:11.000Z",
      "ids": {
        "trakt": 140912,
        "tvdb": 703353,
        "tmdb": 81266
      }
    }
  ],
  "episodes": [
    {
      "watched_at": "2014-09-01T09:10:11.000Z",
      "ids": {
        "trakt": 1061,
        "tvdb": 1555111,
        "imdb": "tt007404",
        "tmdb": 422183
      }
    }
  ]
```
